src/gov/nist/core/LexerCore.java

```diff
@@ -736,7 +736,8 @@ public String number() throws ParseException {
 
         int startIdx = ptr;
         try {
-            if (!isDigit(lookAhead(0))) {
+            char c = lookAhead(0);
+            if (c != '-' && !isDigit(c)) {
```

Comment:
All JUnit parser tests (jain-sip/src/test/unit/gov/nist/javax/sip/parser) are OK.

From the headers above the following are covered:
- HostNameParserTest
- TimeStampParserTest
- ContentLengthParserTest
- CSeqParserTest
- MaxForwardsParserTest
- MimeVersionParserTest
- MinExpiresParserTest
- RetryAfterParserTest

TCK regression is also OK.